scripts/in_container/run_resource_check.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+from dataclasses import dataclass
+from typing import Dict
+
+import psutil
+from rich.console import Console
+
+
+@dataclass
+class Resource:
+    current: int
+    minimumAllowed: int
+
+
+console = Console(force_terminal=True, color_system="standard", width=180)
+
+
+def get_size(kilobytes):
+    """
+    Convert kilobytes into gigabytes
+    1 Gigabytes = 1,048,576 Kb
+    """
+    factor = 1000000000
+    value_gb = kilobytes // factor
+    return value_gb
+
+
+def resoure_check():
+    """
+    Use gsutil to get resources in kylobytes: memory, cpus and disk
+    """
+    MINIMUM_ALLOWED_MEMORY = 4
+    MINIMUM_ALLOWED_CPUS = 2
+    MINIMUM_ALLOWED_DISK = 20
+    print("\nChecking resources.\n")
+
+    # Memory current available
+    svmem = psutil.virtual_memory()
+    mem_available = get_size(svmem.total)
+
+    # Cpus current available
+    cpus_available = psutil.cpu_count(logical=True)
+
+    # Disk current available
+    partitions = psutil.disk_partitions()
+    partition_usage = psutil.disk_usage(partitions[0].mountpoint)
+    disk_available = get_size(partition_usage.free)
+
+    resources: Dict[str, Resource] = {
+        'Memory': Resource(current=mem_available, minimumAllowed=MINIMUM_ALLOWED_MEMORY),
+        'Cpus': Resource(current=cpus_available, minimumAllowed=MINIMUM_ALLOWED_CPUS),
+        'Disk': Resource(current=disk_available, minimumAllowed=MINIMUM_ALLOWED_DISK),
+    }
+    return resources
+
+
+def resoure_validate():
+
+    resources = resoure_check()
+    warning_resources = False
+    check = "OK"
+
+    for resource, capacity in resources.items():
+
+        check = '' if resource == "Cpus" else 'GB'
+
+        if (
```

Comment:
I do not think we need the `resource == "..."` her e at all . This if could lbe reduced to simpy `if capacity.current < capacity.minimumAllowed:`

Reply:
I totally agree! Thank you!